pkg/collector/smbclient/smbclient.go

```diff
@@ -0,0 +1,452 @@
+			} else {
+				return fmt.Errorf("unknown smbclient collector: %s", collectorName)
+			}
+		}
+	}
+
+	return nil
+}
+
+// Collect collects smb client metrics and sends them to prometheus
+func (c *collector) Collect(ctx *types.ScrapeContext, ch chan<- prometheus.Metric) error {
+	collectorFuncs := map[string]func(ctx *types.ScrapeContext, ch chan<- prometheus.Metric) error{
+		"ClientShares": c.collectClientShares,
+	}
+
+	for _, collectorName := range c.enabledCollectors {
+		if err := collectorFuncs[collectorName](ctx, ch); err != nil {
+			_ = level.Error(c.logger).Log("msg", "Error in "+collectorName, "err", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Perflib: SMB Client Shares
+type perflibClientShares struct {
+	Name string
+
+	AvgDataQueueLength                         float64 `perflib:"Avg. Data Queue Length"`
+	AvgReadQueueLength                         float64 `perflib:"Avg. Read Queue Length"`
+	AvgSecPerRead                              float64 `perflib:"Avg. sec/Read"`
+	AvgSecPerWrite                             float64 `perflib:"Avg. sec/Write"`
+	AvgSecPerDataRequest                       float64 `perflib:"Avg. sec/Data Request"`
+	AvgWriteQueueLength                        float64 `perflib:"Avg. Write Queue Length"`
+	CreditStallsPerSec                         float64 `perflib:"Credit Stalls/sec"`
+	CurrentDataQueueLength                     float64 `perflib:"Current Data Queue Length"`
+	DataBytesPerSec                            float64 `perflib:"Data Bytes/sec"`
+	DataRequestsPerSec                         float64 `perflib:"Data Requests/sec"`
+	MetadataRequestsPerSec                     float64 `perflib:"Metadata Requests/sec"`
+	ReadBytesTransmittedViaSMBDirectPerSec     float64 `perflib:"Read Bytes transmitted via SMB Direct/sec"`
+	ReadBytesPerSec                            float64 `perflib:"Read Bytes/sec"`
+	ReadRequestsTransmittedViaSMBDirectPerSec  float64 `perflib:"Read Requests transmitted via SMB Direct/sec"`
+	ReadRequestsPerSec                         float64 `perflib:"Read Requests/sec"`
+	TurboIOReadsPerSec                         float64 `perflib:"Turbo I/O Reads/sec"`
+	TurboIOWritesPerSec                        float64 `perflib:"Turbo I/O Writes/sec"`
+	WriteBytesTransmittedViaSMBDirectPerSec    float64 `perflib:"Write Bytes transmitted via SMB Direct/sec"`
+	WriteBytesPerSec                           float64 `perflib:"Write Bytes/sec"`
+	WriteRequestsTransmittedViaSMBDirectPerSec float64 `perflib:"Write Requests transmitted via SMB Direct/sec"`
+	WriteRequestsPerSec                        float64 `perflib:"Write Requests/sec"`
+}
+
+func (c *collector) collectClientShares(ctx *types.ScrapeContext, ch chan<- prometheus.Metric) error {
+	var data []perflibClientShares
+	if err := perflib.UnmarshalObject(ctx.PerfObjects["SMB Client Shares"], &data, c.logger); err != nil {
+		return err
+	}
+	for _, instance := range data {
+		// labelName := c.toLabelName(instance.Name)
```

Comment:
What should we do about this commented out line and the unused func? I don't see any prometheus restrictions on label values. I looked here https://prometheus.io/docs/practices/naming/.

Reply:
I'm also not aware of any restrictions here.

Reply:
Yes. I am currently running with it commented out.